game/src/contrib/components/ProjectileComponent.java

```diff
@@ -5,35 +5,55 @@
 import core.utils.Point;
 
 /**
- * ProjectileComponent saves the start position and goal location of a projectile.
+ * Marks an entity as a projectile.
+ *
+ * <p>The Component stores a {@link #startPosition start} and a {@link #goalLocation goal}
+ *
+ * <p>The {@link contrib.systems.ProjectileSystem ProjectileSystem} will calculate a flight path
+ * from the start to the goal and will then set the velocity of the entity accordingly. If the goal
```

Comment:
Ist das wirklich so? Sollte ein Projektil nicht eine Geschwindigkeit vom "Auftraggeber" bekommen und das ProjectileSystem muss sich dann um die korrekte Berechnung der Flugbahn sowie der aktuellen Position auf der Flugbahn kümmern?

Reply:
> Sollte ein Projektil nicht eine Geschwindigkeit vom "Auftraggeber" bekommen

Machst du als Auftraggeber ja auch, halt über das VelocityComponent des Projektils. Da gibst du die Geschwindigkeit an.
Dein Vollstäniges Projektil wird am Ende folgendes haben:
- ProjectileComponent
- VelocityComponent
- PositionComponent
- DrawComponent
- und vermutlich ein CollisionComponent (bei sowas wie ein Feuerball)

Reply:
Ah, das hab ich vorhin nicht im Kopf gehabt, wir haben ja die VelocityComponent für die Geschwindigkeit. Hmmm. Im Prinzip müsste oben das "and will then set the velocity of the entity accordingly" erstmal raus - das wird ja nicht über die hiesige ProjectileComponent, sondern die daneben liegende VelocityComponent erreicht. Dazu müsste aber tatsächlich vielleicht noch ein Satz neu mit rein.

Reply:
> Das scheint aus der Dokumentation aber nicht klar zu werden.
Muss mir was überlegen, wie ich das sinnvoll gestalte für Components die eigentlich ohne andere Components keinen Sinn ergeben.

Vielleicht im Klassen-Javadoc der Components jeweils einen kurzen Hinweis, welche Components die aktuelle Component und/oder das verarbeitende System noch braucht. Das kann aber ausufern, oder? Manche Components haben nur ein (Haupt-) System, manche werden von vielen (Neben-) Systems mitgenutzt ... 

Zusätzlich wäre das genau ein Fall für die Doku. Da kann man eine Tabelle basteln. Und hier im Javadoc könnte man einen Link (URL ins Github) auf die Tabelle in der Doku reinpacken.